File: src/MyServiceBus/Serialization/ByteArrayMessageBody.cs
using System.IO;
using System.Text;

namespace MyServiceBus.Serialization;

public sealed class ByteArrayMessageBody : MessageBody
{
    private readonly byte[] _bytes;

    public ByteArrayMessageBody(byte[] bytes)
    {
        ArgumentNullException.ThrowIfNull(bytes);
        _bytes = bytes;
    }

    public long? Length => _bytes.LongLength;

    public Stream GetStream() => new MemoryStream(_bytes, writable: false);

    public byte[] GetBytes() => _bytes;

    public string GetString() => Encoding.UTF8.GetString(_bytes);
}

File: src/Java/myservicebus-abstractions/src/main/java/com/myservicebus/serialization/ByteArrayMessageBody.java
package com.myservicebus.serialization;

import java.io.ByteArrayInputStream;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import java.util.Objects;

public final class ByteArrayMessageBody implements MessageBody {
    private final byte[] bytes;

    public ByteArrayMessageBody(byte[] bytes) {
        this.bytes = Objects.requireNonNull(bytes, "bytes");
    }

    @Override
    public Long getLength() {
        return (long) bytes.length;
    }

    @Override
    public InputStream getStream() {
        return new ByteArrayInputStream(bytes);
    }

    @Override
    public byte[] getBytes() {
        return bytes;
    }

    @Override
    public String getString() {
        return new String(bytes, StandardCharsets.UTF_8);
    }
}
